pkg/monitor/event.go

```diff
@@ -158,8 +213,41 @@ func recordAddOrUpdateEvent(
 	if obj.Type == corev1.EventTypeWarning {
 		condition.Level = monitorapi.Warning
 	}
-	m.RecordAt(t, condition)
 
+	pathoFrom := obj.FirstTimestamp.Time
+	if pathoFrom.IsZero() {
+		pathoFrom = obj.EventTime.Time
+	}
+	if pathoFrom.IsZero() {
+		pathoFrom = obj.CreationTimestamp.Time
+	}
+	to := pathoFrom.Add(1 * time.Second)
+
+	event := monitorapi.EventInterval{
+		Condition: condition,
+	}
+
+	// The matching here needs to mimic what is being done in the synthetictests/testDuplicatedEvents function.
+	eventDisplayMessage := fmt.Sprintf("%s - %s", event.Locator, event.Message)
```

Comment:
Adding the message to the locator would not be practical because that locator is used on the right side of the chart and messages can get very long which would make the chart look bad and not be so useful.

I think (at least for this first iteration), just using the locator is fine because all those events still happened to that object represented by that locator.  The timeline would then show all the interesting/pathological events on one line which seems useful to me.

Reply:
I'm mimicking the comparisons done [here](https://github.com/openshift/origin/blob/2154d73eed665b1a0951f59ee638e52242d64a6c/pkg/synthetictests/duplicated_events.go#L373) so that the chart remains consistent with the `[sig-arch] events should not repeat pathologically...` test.  All of our regexes assume that format.

Reply:
That would then become quite difficult for the etcd team to use to actually debug their pathological repeating event, they'd have to mouseover potentially dozens of 1s vertical lines and try to map out mentally where their actual event was coming from. Your explanation seems reasonable for a general purpose use case, but I do think we need to isolate the specific reasons via locator somehow.

Reply:
Interesting.  I interpret this as two ways to go:

* events differentiated only by locator; multiple events on single line regardless of message.  You can see this in PR CI under these two columns: `bc22087`, `066319b`.
* events differentiated by locator and message (without "n times"); the chart will have a row per locator/message combo.  Multiple events for "event happened n times" will appear on the same line.  We can hash the message without "n times", as you mention above, and add it to the locator so then we'll have a unique locator tied to a specific message

If I got that right and you like the second one better, I can make that happen easily by hashing the message and appending that to the locator string.